src/lib/slide-toggle/slide-toggle.spec.ts

```diff
@@ -355,6 +356,114 @@ describe('MatSlideToggle without forms', () => {
     }));
   });
 
+  describe('custom action configuration', () => {
+    it('should not change value on click when click action is noop', fakeAsync(() => {
+      TestBed
+        .resetTestingModule()
+        .configureTestingModule({
+          imports: [MatSlideToggleModule],
+          declarations: [SlideToggleBasic],
+          providers: [
+            {
+              provide: HAMMER_GESTURE_CONFIG,
+              useFactory: () => gestureConfig = new TestGestureConfig()
+            },
+            { provide: MAT_SLIDE_TOGGLE_DEFAULT_OPTIONS, useValue: {disableToggleValue: true}},
```

Comment:
Extra space (here and below)

Reply:
Should be also able to remove that MutationObserver thing as well.

It's not really needed. Also out of curiosity, why does TestBed.overrideProvider not work?